### PR TITLE
Move all examples to vtctldclient

### PR DESCRIPTION
**Pull request by mattlord**
## Description

[`vtctldclient` is replacing the legacy `vtctlclient`](https://github.com/vitessio/vitess/pull/12866). Starting with the v18 GA (after https://github.com/vitessio/vitess/issues/12152 is done) all users should use `vtctldclient` rather than `vtctlclient`. This PR updates the examples accordingly.

## Related Issue(s)

  - Fixes: https://github.com/vitessio/vitess/issues/14209
  - Part of: https://github.com/vitessio/vitess/issues/12152

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by deepthi — APPROVED** [review]

File: examples/backups/stop_tablets.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ for tablet in 100 200 300; do
       echo "Shutting down mysql zone1-$uid"
       CELL=zone1 TABLET_UID=$uid ../common/scripts/mysqlctl-down.sh
       echo "Removing tablet directory zone1-$uid"
-      vtctlclient DeleteTablet -- --allow_primary=true zone1-$uid
+      vtctldclient DeleteTablets --allow-primary=true zone1-$uid
```
Comment (ajm188) [inline examples/backups/stop_tablets.sh:32]: TIOLI: `DeleteTablets --allow-primary zone1-$uid` (no explicit `=true`) also works

File: examples/compose/externaldb_vreplication.sh [hunk 2]
```diff
@@ -43,33 +43,27 @@ dest_tablet=$(echo "$TABLET_INFO "| grep -v "ext_" | grep "primary" | awk '{ pri
...
+/vt/bin/vtctldclient --server $VTCTLD_SERVER ExecuteFetchAsDba $dest_alias 'SET GLOBAL FOREIGN_KEY_CHECKS=0;'
 
 # Get source_sql mode
-source_sql_mode=$(/vt/bin/vtctlclient --server $VTCTLD_SERVER ExecuteFetchAsDba $source_alias 'SELECT @@GLOBAL.sql_mode' | awk 'NR==4 {print $2}')
+source_sql_mode=$(/vt/bin/vtctldclient --server $VTCTLD_SERVER ExecuteFetchAsDba $source_alias 'SELECT @@GLOBAL.sql_mode' | awk 'NR==4 {print $2}')
 
 # Apply source sql_mode to destination
 # The intention is to avoid replication errors
-/vt/bin/vtctlclient --server $VTCTLD_SERVER ExecuteFetchAsDba $dest_alias "SET GLOBAL sql_mode='$source_sql_mode';"
+/vt/bin/vtctldclient --server $VTCTLD_SERVER ExecuteFetchAsDba $dest_alias "SET GLOBAL sql_mode='$source_sql_mode';"
```
Comment (ajm188) [inline examples/compose/externaldb_vreplication.sh:53]: technically this is an alias for the actual command, [`ExecuteFetchAsDBA`](https://github.com/vitessio/vitess/blob/main/go/cmd/vtctldclient/command/query.go#L40-L48)

File: examples/compose/vtcompose/vtcompose.go [hunk 3]
```diff
@@ -534,7 +534,7 @@ func generateDefaultShard(tabAlias int, shard string, keyspaceData keyspaceInfo,
   path: /services/init_shard_primary%[2]d
   value:
     image: vitess/lite:${VITESS_TAG:-latest}
-    command: ["sh", "-c", "/vt/bin/vtctlclient %[5]s InitShardPrimary -force %[4]s/%[3]s %[6]s-%[2]d "]
+    command: ["sh", "-c", "/vt/bin/vtctldclient %[5]s InitShardPrimary --force %[4]s/%[3]s %[6]s-%[2]d "]
```
Comment (ajm188) [inline examples/compose/vtcompose/vtcompose.go:537]: simply `-f` also works if we want to demo that in the examples

**Review by ajm188 — APPROVED** [review]

**Review by frouioui — APPROVED** [review]
LGTM! We should probably update the documentation as well to only use `vtctldclient`

**Comment by frouioui** [comment]
> We should probably update the documentation as well to only use `vtctldclient`

This is not an explicit list but I am seeing different mentions to `vtctlclient`:
- https://vitess.io/docs/18.0/get-started/operator/#setup-port-forward
- https://vitess.io/docs/18.0/user-guides/migration/move-tables/#start-the-move
- https://vitess.io/docs/18.0/user-guides/migration/move-tables/#monitoring-progress-optional
- https://vitess.io/docs/18.0/user-guides/migration/move-tables/#validate-correctness-optional
- https://vitess.io/docs/18.0/user-guides/migration/move-tables/#switching-traffic
- https://vitess.io/docs/18.0/user-guides/migration/move-tables/#reverting-the-switch-optional

File: examples/compose/vtcompose/vtcompose.go (hunk 3 above)
Comment (mattlord) [reply examples/compose/vtcompose/vtcompose.go:537]:
```
vtctldclient InitShardPrimary -f
E1010 20:01:59.194594   92668 main.go:56] unknown shorthand flag: 'f' in -f
```

Comment (ajm188) [reply examples/compose/vtcompose/vtcompose.go:537]: ah shoot, sorry. we _should_ make that work (later)

**Pull request merged**